syft/pkg/cataloger/javascript/parse_package_lock.go

```diff
@@ -61,7 +80,9 @@ func parsePackageLock(resolver source.FileResolver, _ *generic.Environment, read
 
 	if lock.LockfileVersion == 1 {
 		for name, pkgMeta := range lock.Dependencies {
-			pkgs = append(pkgs, newPackageLockV1Package(resolver, reader.Location, name, pkgMeta))
+			canonicalName, canonicalVersion := resolvePackageLockNameAliases(name, pkgMeta.Version)
```

Comment:
Since this is _just_ for v1, it might be nice to just move it into the `newPackageLockV1Package` function instead of doing the `resolvePackageLockNameAliases` here

Reply:
this sounds reasonable thanks.

Have moved this logic to `newPackageLockV1Package`.

Have also removed the tests for the now deleted `resolvePackageLockNameAliases`, however the same coverage is acheived with the other tests introduced in this PR.